conan/internal/model/conf.py

```diff
@@ -384,6 +384,12 @@ def copy(self):
         c._values = OrderedDict((k, v.copy()) for k, v in self._values.items())
         return c
 
+    def filter_core(self):
+        c = Conf()
+        c._values = OrderedDict((k, v.copy()) for k, v in self._values.items()
+                                if not k.startswith("core"))
```

Comment:
Should we use the `CORE_CONF_PATTERN` instead?

```suggestion
                                if not CORE_CONF_PATTERN.match(k))
```